### PR TITLE
feat: add bprotocol-liquity deposit and withdraw actions and tests

### PR DESCRIPTION
**Pull request by kjaredic**

### CODE REVIEW

File: contracts/actions/bprotocol/BprotocolLiquitySPDeposit.sol [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import "../../interfaces/bprotocol/IBAMM.sol";
+import "../ActionBase.sol";
+import "./helpers/BprotocolLiquitySPHelper.sol";
+import "../../utils/TokenUtils.sol";
+
+
+contract BprotocolLiquitySPDeposit is ActionBase, BprotocolLiquitySPHelper {
+    using TokenUtils for address;
+
+    struct Params {
```
Comment (yippee-ki-yay) [inline contracts/actions/bprotocol/BprotocolLiquitySPDeposit.sol:13]: Usually we do the amount variable first and the from/to last, switch places to keep variable order consistent between actions

Comment (yippee-ki-yay) [reply contracts/actions/bprotocol/BprotocolLiquitySPDeposit.sol:13]: Also add natspec for stuff like to explain that during deposit lqty rewards are sent to the proxy

File: contracts/actions/bprotocol/BprotocolLiquitySPWithdraw.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import "../../utils/TokenUtils.sol";
+
+
+contract BprotocolLiquitySPWithdraw is ActionBase, BprotocolLiquitySPHelper {
+    using TokenUtils for address;
+
+    struct Params {
+        address to;
+        address lqtyTo;
+        uint256 amount;
```
Comment (yippee-ki-yay) [inline contracts/actions/bprotocol/BprotocolLiquitySPWithdraw.sol:16]: amount is non specific, given the user might think it's lusdAmount and it's lpAmount/shareAmount

File: contracts/actions/bprotocol/BprotocolLiquitySPWithdraw.sol [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        bytes32[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+        params.to = _parseParamAddr(params.to, _paramMapping[0], _subData, _returnValues);
+        params.lqtyTo = _parseParamAddr(params.lqtyTo, _paramMapping[1], _subData, _returnValues);
+        params.amount = _parseParamUint(params.amount, _paramMapping[2], _subData, _returnValues);
+
+        (uint256 deposited, bytes memory logData) = _withdraw(params);
```
Comment (yippee-ki-yay) [inline contracts/actions/bprotocol/BprotocolLiquitySPWithdraw.sol:30]: `uint256 deposited` instead of lusdWithdrawn

**Review by yippee-ki-yay — CHANGES_REQUESTED** [review]
Needs a few things to be cleaned up, but overall seems functional.
One thing to also think about is the ability to send lpTokens to a dff. address after deposit and in withdraw pull from a diff. address, something like we have in Yearn.

File: contracts/actions/bprotocol/BprotocolLiquitySPDeposit.sol (hunk 1 above)
Comment (kjaredic) [reply contracts/actions/bprotocol/BprotocolLiquitySPDeposit.sol:18]: will do

**Review by yippee-ki-yay — APPROVED** [review]

**Pull request merged**